.github/workflows/docker-publish.yml

```diff
@@ -0,0 +1,64 @@
+name: Publish on Github container registry
+
+on:
+  release:
+    type: [published]
+  push:
+    branches:
+      - main
+      - feat/ci_docker_build
+  pull_request:
+    paths:
+      - 'pyproject.toml'
+      - 'Dockerfile'
```

Comment:
Re Q1: I think the `on: push: branches: main` rule should ensure that the workflow is always run upon merging, no? (https://github.com/orgs/community/discussions/27017)

Re Q2: Setting this scope was a bit of a "lazy" compromise, because it only includes the minimal scope and is not really sufficient for our use case as the `client image` contains the full modo api and the `server image` depends on it. E.g. when I added the `s3_endpoint` attribute to `MODO` the client and the also server image were outdated and their functionality broke.
We could completely remove the scope to any path and use this instead:
```
on:
  workflow_dispatch
  release:
    type: [published]
  push:
    branches:
      - main
  pull_request:
    types:
       - opened
       - reopened
```
This should ensure that we get an updated image for each PR (makes it easier to test and review), release, push to main and gives us a manual trigger?

Reply:
Q1: indeed, looks good :+1: 
Q2: I think so yes, your suggestion seems to be what we want, and in doubt, we can always trigger image builds manually during the PR before merging